### PR TITLE
Update upstream to mariadb 10.2.14

### PR DESCRIPTION
**Pull request by fritzm**

### CODE REVIEW

File: ups/eupspkg.cfg.sh [hunk 1]
```diff
@@ -6,37 +6,26 @@ config()
...
     ARGS+=("-DCMAKE_INSTALL_PREFIX=${PREFIX}")
+
+    # Remove mysql-test directory (~330 MB) to reduce installation size
+    ARGS+=("-DINSTALL_MYSQLTESTDIR=")
+
     # TokuDB requires third-party software jemalloci (see http://www.canonware.com/jemalloc/)
     ARGS+=('-DPLUGIN_TOKUDB=NO')
 
     # Due to cmake library discovery being overly energetic (searching every
     # lib directory relative to each entry in $PATH) we use the bundled SSL and
```
Comment (timj) [inline ups/eupspkg.cfg.sh]: Comment now out of date.

File: ups/eupspkg.cfg.sh [hunk 2]
```diff
@@ -6,37 +6,26 @@ config()
     ARGS=()
 
     ARGS+=("-DCMAKE_INSTALL_PREFIX=${PREFIX}")
+
+    # Remove mysql-test directory (~330 MB) to reduce installation size
+    ARGS+=("-DINSTALL_MYSQLTESTDIR=")
```
Comment (timj) [inline ups/eupspkg.cfg.sh:11]: I'm surprised we didn't do this when we made the same change to mariadb package.

**Review by timj — APPROVED** [review]

File: ups/eupspkg.cfg.sh [hunk 3]
```diff
@@ -6,37 +6,26 @@ config()
...
-        *)
-            # non-fatal
-            echo "unsupported platform: $(uname)"
-            ;;
-    esac
+    # Prevent CMake from finding and linking against libraries distributed in $(dirname
+    # python)/../lib. This is CMake's default behavior, but can cause us to erroneously link
+    # against libraries distributed by e.g. Anaconda.
+    PYTHONLIBDIR=$(which python | sed -e's|bin/python|lib|')
+    ARGS+=("-DCMAKE_SYSTEM_IGNORE_PATH=${PYTHONLIBDIR}")
```
Comment (jhoblitt) [inline ups/eupspkg.cfg.sh:26]: Why is this needed?

Comment (timj) [reply ups/eupspkg.cfg.sh:26]: For the same reason it was already in `mariadb`. If cmake finds a library in the Anaconda lib then all the builds fail because nothing else is looking in that library directory. This lets us mask out Anaconda from the search path.

Comment (jhoblitt) [reply ups/eupspkg.cfg.sh:26]: Can you give me an example of that happening?  I've built the entire stack with this removed.

Comment (timj) [reply ups/eupspkg.cfg.sh:26]: All of the examples that caused us to add it in the first place. The problem is that *sometimes* someone puts a library in their conda installation that we aren't expecting and then everything breaks. In your case you probably hadn't installed any extra packages. DM-5595 was the relevant Jira ticket. In that case it was `libz` (which we now bypass by specifically asking for the internal one) but in general we don't want cmake to ever find any library in the conda lib path.
